internal/client/eth_client.go

```diff
@@ -23,31 +30,84 @@ type EthClientInterface interface {
 	BlockByNumber(ctx context.Context, number *big.Int) (*types.Block, error)
 }
 
-// NewEthClient ... Initializer
-func NewEthClient() EthClientInterface {
+func NewEthClient(restyClient *resty.Client) EthClientInterface {
+	if restyClient == nil {
+		restyClient = resty.New()
+	}
+
+	restyDebug, _ := strconv.ParseBool(os.Getenv("RESTY_DEBUG"))
```

Comment:
I'd rather us not load env vars through `os.Getenv` and instead comply with existing [config](https://github.com/base-org/pessimism/blob/master/internal/config/config.go) handling paradigm of loading all process level env vars within the same construct. To achieve this here, we'd need to add a `EthClientCfg` type that persists as a data member of the `EthClient` struct.

Reply:
+1